### PR TITLE
Support helper

### PR DESCRIPTION
**Pull request by itdove**
https://github.com/open-cluster-management/library-go/issues/17

### CODE REVIEW

File: docs/applier.md [hunk 1]
```diff
@@ -1,20 +1,22 @@
 # Introduction
 
 The file [applier](../pkg/applier) contains an number of methods allowing you to render template yamls. 
-The resources are read by an Go object satisfying the [TemplateReader](pkg/applier/templateProcessor.go) reader.  
+The template support the [text/template](https://golang.org/pkg/text/template/) functions, the [Mastermind/sprig](https://github.com/Masterminds/sprig) functions and you can develop you own functions too [templatefunction.go](../pkg/applier/templatefunction.go).
+A `_helpers.tpl` file can also be included to define your own functions.
```
Comment (hanqiuzh) [inline docs/applier.md:5]: This description is not clear, own functions? Or just for {{define}}

File: pkg/applier/templateprocessor.go [hunk 2]
```diff
@@ -117,11 +144,20 @@ func (tp *TemplateProcessor) TemplateResource(
 	templateName string,
 	values interface{},
 ) ([]byte, error) {
+	klog.V(5).Infof("templateName: %s", templateName)
+	if filepath.Base(templateName) == "_helpers.tpl" {
```
Comment (hanqiuzh) [inline pkg/applier/templateprocessor.go:148]: _helpers.tpl is skipped here, is there any other functions to load it?

File: pkg/applier/applier.go [hunk 3]
```diff
@@ -42,19 +54,32 @@ func NewApplier(
...
 		return nil, goerr.New("applier is nil")
 	}
 	if client == nil {
 		return nil, goerr.New("client is nil")
 	}
+	if applierOptions == nil {
+		applierOptions = &ApplierOptions{}
+	}
+	if applierOptions.Backoff == nil {
+		applierOptions.Backoff = &wait.Backoff{
```
Comment (hanqiuzh) [inline pkg/applier/applier.go:69]: not sure, so what's the worse case when fail to apply, 100,500,2500,12500,72500 ms? shall we add a cap here? and not sure will it block reconcile or it only used in executable? feel in reconcile we want method not blocking but reply at once

Comment (itdove) [reply pkg/applier/applier.go]: Replaced by retry.defaultBackoff and add a custom one in the cli.

**Review by hanqiuzh — DISMISSED** [review]

**Pull request merged**